src/librustc_parse/parser/ty.rs

```diff
@@ -276,8 +276,33 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_borrowed_pointee(&mut self) -> PResult<'a, TyKind> {
-        let opt_lifetime = if self.check_lifetime() { Some(self.expect_lifetime()) } else { None };
+        let mut opt_lifetime =
+            if self.check_lifetime() { Some(self.expect_lifetime()) } else { None };
         let mutbl = self.parse_mutability();
+        if mutbl == Mutability::Mut && opt_lifetime.is_none() && self.token.is_lifetime() {
```

Comment:
```suggestion
        if self.token.is_lifetime() && mutbl == Mutability::Mut && opt_lifetime.is_none()  {
```
nit: probably irrelevant perf improvement. :laughing: